main/src/library/Graph.flix

```diff
@@ -31,23 +31,51 @@ namespace Graph {
         };
         let x = query edges, reachability select (src, dst) from Reachable(src, dst);
         List.toSet(x)
-    }
 
     ///
-    /// Returns the vertices that are reachable from the `origin` in the directed graph `g`.
+    /// Returns the vertices that are reachable from the `src` in the directed graph `g`.
     ///
-    pub def reachable(origin: t, g: m[(t, t)]): Set[t] with Foldable[m], Boxable[t] = {
+    pub def reachable(src: t, g: m[(t, t)]): Set[t] with Foldable[m], Boxable[t] =
         let edges = inject g into Edge;
         let reachability = #{
             // A node can reach itself.
-            Reachable(origin).
-            // If `origin` can reach `m` and there is an edge from `m` to `n`
-            // then `origin` can also reach `n`. This adds all node pairs to
+            Reachable(src).
+            // If `src` can reach `m` and there is an edge from `m` to `n`
+            // then `src` can also reach `n`. This adds all node pairs to
             // the relational that are reachable using any number of nodes.
             Reachable(n) :- Reachable(m), Edge(m, n).
         };
         let x = query edges, reachability select dst from Reachable(dst);
         List.toSet(x)
-    }
+
+    ///
+    /// Returns the vertices that are unreachable from the `src` in the directed graph `g`.
+    ///
+    pub def unreachable(src: t, g: m[(t, t)]): Set[t] with Foldable[m], Boxable[t] =
+        let edges = inject g into Edge;
+        let nodes = #{
+            Node(x) :- Edge(x, _).
```

Comment:
I was wondering about src and dst that are not nodes in any edge and it seems to work fine for all of them. Here `Node(src)` should technically be a fact but it doesn't influence the output. Maybe there should be tests for these cases too?

Reply:
Good idea. I didn't even think of that

Reply:
Right now `Graph.reachable(1, Set#{}: Set[(Int32, Int32)])` returns `Set#{1}`. Should it return `Set#{}` instead?

Reply:
I think that is correct. The initial node is reachable.